py_bind/optv/parameters.pyx

```diff
@@ -0,0 +1,58 @@
+#Implementation of Python binding to parameters.h
+from libc.stdlib cimport malloc, free
+
+cdef class Py_mm_np:
+      
+    def __init__(self, **kwargs):
+        if len(kwargs.keys()) == 0:
+            self._owns_data = 0
+            return
+        
+        self._owns_data = 1
+        self._mm_np = <mm_np *>malloc(sizeof(mm_np))
+        
+        self._mm_np[0].nlay = kwargs['nlay']
+        self._mm_np[0].n1 = kwargs['n1']
+
+        i=0 # setting n2
+        while i<len(kwargs['n2']):
```

Comment:
More Pythonic:
``for i in range(len(...)):``
Then you don't need the i = 0 stuff.